### DISCUSSION TITLE
Export/Import

### DISCUSSION THREAD

**Discussion by IlanZ93** [webextensions/live-css-editor#112] [Q&A] [answered]
Hello, 

Is it possible to Export/Import all the data ? 

Im goin to change my computer, but i dont remember every css i've made so it could be great if i can export data and import on my new computer. I dont use Google Sync because i have a lot of shit on this i have to clean it first, one day.

Thank you

**Comment by webextensions** [comment, marked as answer]
@IlanZ93 
For the users who are familiar with JS, they can copy the raw data with a script.

* Load the extension on a webpage
* Open browser DevTools
* Go to the Console tab in DevTools
* Select the "JavaScript execution context" as Magic CSS (by default, the context would be "top")
* Run the following script:
```js
var copyToClipboard=copy;
chrome.storage.local.get(null,function(val){
    copyToClipboard(JSON.stringify(val,null,'  '));
});
```
This would copy the CSS data to the clipboard.

Note: It doesn't include the global configuration/settings.

Now, in the other machine:
* Install the extension
* Load the extension on a webpage
* Open browser DevTools
* Go to the Console tab in DevTools
* Select the "JavaScript execution context" as Magic CSS (by default, the context would be "top")
* Run the following script:
```js
var data = { ... }; // The copied data goes here
chrome.storage.local.set(data,function(val){
    console.log('done');
});
```

Note: If one wishes to use the "Pin" feature ("Apply styles automatically"), then it may be required to do unpin followed by a pin (to request for the appropriate permissions).

In upcoming updates, the feature of export/import would be available in a user-friendly manner.

**Reply by IlanZ93** [reply]
Thank you for the explanations ! 🙏🏼
